LongWarOfTheChosen/Src/LW_FactionBalance/Classes/X2Ability_TemplarAbilitySet_LW.uc

```diff
@@ -603,6 +611,175 @@ static function FocusLevelModifiers CreateFocusLevelModifiers(int DodgeBonus, in
+
+	return Template;
+}
+
+
+
+static function EventListenerReturn VoltFocusKillTracker_Listener(Object EventData, Object EventSource, XComGameState GameState, Name EventID, Object CallbackData)
+{
+	local XComGameStateContext_Ability AbilityContext;
+	local XComGameStateContext FindContext;
+	local int VisualizeIndex;
+	local XComGameStateHistory History;
+	local XComGameState_Ability Ability;
+	local XComGameState_Unit SourceUnit;
+
+	History = `XCOMHISTORY;
+
+	FindContext = GameState.GetContext();
+	AbilityContext = XComGameStateContext_Ability(GameState.GetContext());
+
+	SourceUnit = XComGameState_Unit(History.GetGameStateForObjectID(AbilityContext.InputContext.SourceObject.ObjectID));
+
+	Ability = XComGameState_Ability(History.GetGameStateForObjectID(SourceUnit.FindAbility('VoltFocusKillTracker', AbilityContext.InputContext.ItemObject).ObjectID));
+	if (AbilityContext != None && AbilityContext.InterruptionStatus != eInterruptionStatus_Interrupt)
+	{
+		if (AbilityContext.InputContext.AbilityTemplateName == 'Volt')
+		{
+			VisualizeIndex = GameState.HistoryIndex;
+			while( FindContext.InterruptionHistoryIndex > -1 )
+			{
+				FindContext = History.GetGameStateFromHistory(FindContext.InterruptionHistoryIndex).GetContext();
+				VisualizeIndex = FindContext.AssociatedState.HistoryIndex;
+			}
+			Ability.AbilityTriggerAgainstSingleTarget(SourceUnit.GetReference(), false, VisualizeIndex);
+		}
+	}
+	return ELR_NoInterrupt;
+}
+
+
+static function X2AbilityTemplate DoubleRendFocus()
+{
+	local X2AbilityTemplate Template;
+	local X2Effect_ModifyTemplarFocus FocusEffect;
+	local X2AbilityTrigger_EventListener EventListener;
+
+	`CREATE_X2ABILITY_TEMPLATE(Template, 'DoubleRendFocus');
+	Template.IconImage = "img:///UILibrary_PerkIcons.UIPerk_advent_commandaura";
+	Template.Hostility = eHostility_Neutral;
+//BEGIN AUTOGENERATED CODE: Template Overrides 'Overcharge'
+	Template.AbilitySourceName = 'eAbilitySource_Psionic';
+//END AUTOGENERATED CODE: Template Overrides 'Overcharge'
+	Template.eAbilityIconBehaviorHUD = eAbilityIconBehavior_NeverShow;
+
+	Template.AdditionalAbilities.AddItem('DoubleRendFocusPassive');
+	
+	Template.AbilityToHitCalc = default.DeadEye;
+	Template.AbilityTargetStyle = default.SelfTarget;
+
+	EventListener = new class'X2AbilityTrigger_EventListener';
+	EventListener.ListenerData.Deferral = ELD_OnStateSubmitted;
+	EventListener.ListenerData.EventFn = class'XComGameState_Ability'.static.AbilityTriggerEventListener_Self;
+	EventListener.ListenerData.EventID = 'RendActivated';
+	EventListener.ListenerData.Filter = eFilter_Unit;
+	Template.AbilityTriggers.AddItem(EventListener);
+
+	//	Chance to grant focus when activating Rend if the Templar has Overcharge
+	FocusEffect = new class'X2Effect_ModifyTemplarFocus';
+	FocusEffect.TargetConditions.AddItem(new class'X2Condition_GhostShooter');
+	FocusEffect.ApplyChance = default.OVERCHARGE_FOCUS_CHANCE;
+	Template.AddShooterEffect(FocusEffect);
```

Comment:
The ability is based on overcharge which grants rend focus on everything but kill. This is basically overcharge, without the not kill restriction. but sure, it could be a lot clearer.

Reply:
So it grants +1 focus on hits and kills, so if you kill you get +2 focus?

Reply:
Yes